chartiq/src/main/java/com/chartiq/sdk/scriptmanager/ChartIQScriptManager.kt

```diff
@@ -241,9 +241,9 @@ internal class ChartIQScriptManager : ScriptManager {
     override fun getSetChartPropertyScript(property: String, value: Any): String {
         val safeProperty = property.asSafeScriptParameter
         return if (value is String) {
-            MOBILE_BRIDGE_NAME_SPACE + "setChartProperty(\"${safeProperty}\", \"${value.asSafeScriptParameter}\");"
+            CHART_IQ_JS_OBJECT + "chart.${safeProperty} = \"${value.asSafeScriptParameter}\";"
```

Comment:
What was wrong with using the bridge and setChartProperty to set the value?

Reply:
Both the `setEngineProperty` and `setChartProperty` bridge functions are flawed. They work great for fields that are one level, but if someone wants to change a value in a multi level object (`layout`) it won't work. The functions need to be reworked so they are able to take something like "layout.periodicity" to `setEngineProperty`. `setChartProperty` likely doesn't have as much issue as `setEngineProperty`, but it'll be nice to get them to match.

I created a ticket a long time ago for refactoring, but I'm not seeing it in Kanbanize anymore. It might of been deleted.